px4_ros2_cpp/include/px4_ros2/control/setpoint_types/experimental/attitude.hpp

```diff
@@ -27,9 +27,13 @@ class AttitudeSetpointType : public SetpointBase
   float desiredUpdateRateHz() override {return 200.f;}
 
   void update(
-    const Eigen::Quaternionf & attidude_setpoint,
+    const Eigen::Quaternionf & attitude_setpoint,
     const Eigen::Vector3f & thrust_setpoint_frd, float yaw_sp_move_rate_rad_s = 0.f);
 
+  void update(
```

Comment:
```suggestion
  /**
   * @brief Send an attitude setpoint to the flight controller.
   * Euler angles follow the RPY extrinsic Tait-Bryan convention (equiv. YPR intrinsic)
   *
   * @param roll Attitude setpoint roll angle [rad]
   * @param pitch Attitude setpoint pitch angle [rad]
   * @param yaw Attitude setpoint yaw angle [rad]
   * @param thrust_setpoint_body Throttle demand [-1, 1]^3
   * @param yaw_sp_move_rate_rad_s Yaw setpoint move rate [rad/s]
  */
  void update(
```

I think it might be worth documenting this method overload to specify which euler angle conventions we use to avoid confusion.

If we want to be consistent with the PX4 matrix lib, it uses [3-2-1 intrinsic Tait-Bryan](https://github.com/PX4/PX4-Autopilot/blob/8dae9905aa4ac294dca9eead9459cfddb9094072/src/lib/matrix/matrix/Euler.hpp#L7) euler angles i.e. the rotation order is yaw, pitch, roll and the axes of rotations change with each consecutive rotation (equivalent to 1-2-3 extrinsic).

Using Eigen, this would be:
```
  Eigen::Quaternionf att_setpoint_q{
    Eigen::AngleAxisf(yaw, Eigen::Vector3f::UnitZ()) *
    Eigen::AngleAxisf(pitch, Eigen::Vector3f::UnitY()) *
    Eigen::AngleAxisf(roll, Eigen::Vector3f::UnitX())};
```
(as Beat mentioned, identical to the util function being added)